pkg/pillar/types/zedroutertypes.go

```diff
@@ -128,6 +128,50 @@ func (status AppNetworkStatus) VerifyFilename(fileName string) bool {
 			fileName, expect)
 	}
 	return ret
+
+}
+
+// PerInterfaceErrorMap - Used to return per-interface errors.
+//  ifName is usually used as the key
+type PerInterfaceErrorMap struct {
+	IntfErrorMap map[string]ErrorInfo
+}
+
+// AddError - Add an error into PerInterfaceErrorMap. If en entry already
+//  exists for the interface, it is appended
+func (intfMap *PerInterfaceErrorMap) AddError(ifName string,
+	err error, source string) {
+	errInfo, ok := intfMap.IntfErrorMap[ifName]
+	if ok {
+		errInfo.Error = errInfo.Error + " \n " + err.Error()
```

Comment:
Are you sure this gets called multiple times for the same interface? If not there is no need to concat the error strings.

Reply:
Right now - there is the case in VerifyDeviceNetworkStatus where when there could be an error in Check proxy - and if we don't return, we can see errors in VerifyAllIntfs as well.. That is where this is currently used.

So this addition is currently used in case of AddMap - Whenever we have errors coming from 2 different checks for the same interfaces.. The changes for the checkProxy code triggered this. But it is reasonable to assume we will get more than one error on the interface.

Please note, the connectivity can succeed, but one of the configured interfaces can still have errors.. And this allows all the errors to be detected and sent back.. not just the last or the first error.